source/sigv4.c

```diff
@@ -1812,31 +1833,39 @@ static SigV4Status_t generateCredentialScope( const SigV4Parameters_t * pSigV4Pa
                     /* A field should never be empty, but a value can be empty
                      * provided a field was specified first. */
                 }
+                /* If the previous parameter did not have a value, write the name and empty value for it. */
                 else if( !fieldHasValue )
                 {
-                    /* Store information about Query Parameter Key in the canonical context. This query parameter does not have an associated value. */
+                    /* Store information about previous query parameter name. The query parameter has no associated value. */
                     setQueryParameterKey( currentParameter, &pQuery[ startOfFieldOrValue ], i - startOfFieldOrValue, pCanonicalRequest );
 
-                    /* The previous field did not have a value set for it, so set its value to NULL. */
+                    /* Store the previous parameter's empty value information. Use NULL to represent empty value. */
                     setQueryParameterValue( currentParameter, NULL, 0U, pCanonicalRequest );
 
                     startOfFieldOrValue = i + 1U;
                     currentParameter++;
+
+                    /* Reset the parameter value state of the new parameter entry. */
+                    fieldHasValue = false;
```

Comment:
This looks like a no op to me. Why have it?

Reply:
Removed